lib/adf_diag.py

```diff
@@ -915,4 +933,25 @@ def setup_run_cvdp(self):
         print('For CVDP information visit: https://www.cesm.ucar.edu/working_groups/CVC/cvdp/')
         print('   ')
 
+    #########
+
+    def derive_variables(self,vars_to_derive=None,ts_dir=None):
+
+        """
+        Derive variables acccording to steps given here.  Since derivations will depend on the 
+        variable, each variable to derive will need its own set of steps below.
+        """
+
+        for var in vars_to_derive:
+            if var == "PRECT":
+                 # PRECT can be found by simply adding PRECL and PRECC
+                 # grab file names for the PRECL and PRECC files from the case ts directory
+                 constit_files = sorted(glob.glob(ts_dir+"/*PREC*"))
```

Comment:
I believe you'll want to do the following here in order to avoid potential non-POSIX OS issues:

```suggestion
                 constit_files = sorted(glob.glob(os.path.join(ts_dir,"*PREC*")))
```

Reply:
Done.